pylint/checkers/base/name_checker/checker.py

```diff
@@ -545,46 +560,67 @@ def _assigns_typevar(node: Optional[nodes.NodeNG]) -> bool:
                 return True
         return False
 
-    def _check_typevar_variance(self, name: str, node: nodes.AssignName) -> None:
-        """Check if a TypeVar has a variance and if it's included in the name.
-
-        Returns the args for the message to be displayed.
-        """
+    def _check_typevar(self, name: str, node: nodes.AssignName) -> None:
+        """Check for TypeVar lint violations."""
         if isinstance(node.parent, nodes.Assign):
             keywords = node.assign_type().value.keywords
         elif isinstance(node.parent, nodes.Tuple):
             keywords = (
                 node.assign_type().value.elts[node.parent.elts.index(node)].keywords
             )
 
+        variance = TypeVarVariance.invariant
         for kw in keywords:
-            if kw.arg == "covariant" and kw.value.value:
-                if not name.endswith("_co"):
-                    suggest_name = f"{re.sub('_contra$', '', name)}_co"
-                    self.add_message(
-                        "typevar-name-incorrect-variance",
-                        node=node,
-                        args=(name, "covariant", suggest_name),
-                        confidence=interfaces.INFERENCE,
-                    )
-                return
-
-            if kw.arg == "contravariant" and kw.value.value:
-                if not name.endswith("_contra"):
-                    suggest_name = f"{re.sub('_co$', '', name)}_contra"
-                    self.add_message(
-                        "typevar-name-incorrect-variance",
-                        node=node,
-                        args=(name, "contravariant", suggest_name),
-                        confidence=interfaces.INFERENCE,
-                    )
-                return
+            if variance == TypeVarVariance.double_variant:
+                pass
```

Comment:
```suggestion
                break
```

I guess?

Reply:
This should be `pass`. The reason makes more sense if you look at the follow up PR https://github.com/jpy-git/pylint/pull/1/files

Basically to check the typevar name we need to scan all keyword args since a user could feasibly write
```python
_T = TypeVar(covariant=True, contravariant=True, name="_S")
```
If we broke on discovering double variance then we wouldn't find the name in the last kwarg 😄